### ISSUE TITLE
Question: API to print a structure

### ISSUE THREAD

**Issue by cancerberoSgx** [dsherret/ts-simple-ast#490]
Is there a public API to print an Structure as a string without having to touch the AST or create a source file ? Thanks!

**Comment by dsherret** [comment]
Nope, not at the moment. The printing of structures can be dependent on the context of where they're printed. Could you provide a use case for how this would be useful beyond what's currently possible?

**Comment by cancerberoSgx** [comment]
The use case would be, In a LanguageService Plugin or when interacting with an existing TS Project/Program instance, if there is no need to create a ts-simple-ast project but want to add a new declaration to an existing SourceFile using ts-simple-ast structures for this (since they are much more easy to use than current TypeScript node creation API). Currently I need to instantiate a new project, create a new sourcefile and add the declaration structure there, then obtain the new node's text. This doesn't add too much overhead but it would be ideal to just print the structure as string given UserPreferences and FormatCodeSettings and perhaps initial indentation
